Add support for external spec file via the GCC_SPECS env var.  This 
allows us to easily control pie/ssp defaults with gcc-config profiles.

Patch by Rob Holland.

--- gcc-3/gcc/gcc.c
+++ gcc-3/gcc/gcc.c
@@ -6421,6 +6421,21 @@
 
   /* Process any user specified specs in the order given on the command
      line.  */
+
+  GET_ENVIRONMENT (specs_file, "GCC_SPECS");
+  if (specs_file && (strlen(specs_file) > 0))
+    {
+      struct user_specs *user = (struct user_specs *)
+        xmalloc (sizeof (struct user_specs));
+
+      user->next = (struct user_specs *) 0;
+      user->filename = specs_file;
+      if (user_specs_tail)
+        user_specs_tail->next = user;
+      else
+        user_specs_head = user;
+      user_specs_tail = user;
+    }
   for (uptr = user_specs_head; uptr; uptr = uptr->next)
     {
       char *filename = find_a_file (&startfile_prefixes, uptr->filename,
